pkg/clustermanager/manager.go

```diff
@@ -400,6 +397,10 @@ func VerifyIgnoreDNSName(caCertsPEM []byte) (func(rawCerts [][]byte, verifiedCha
 }
 
 func (m *Manager) toRecord(ctx context.Context, cluster *v3.Cluster) (*record, error) {
+	if cluster.DeletionTimestamp != nil {
```

Comment:
Same comment, but this is only called from start so not really needed either way.